File: By-Swift/By-Swift/Programmers/스택,큐/짝지어제거하기.swift
//
//  짝지어제거하기.swift
//  Algorithm-Swift
//
//  Created by 이원석 on 2021/05/23.

import Foundation

func solution(_ s:String) -> Int{
    var arr = Array(s)
    var temp = [Character]()
    
    for i in arr.indices {
        if temp.isEmpty {
            temp.append(arr[i])
        } else {
            if temp.last! == arr[i] {
                temp.removeLast()
            } else {
                temp.append(arr[i])      
            }
        }
    }

    return temp.isEmpty ? 1 : 0
}

File: By-Python/programmers/[스택,큐]/짝지어제거하기.py
# 짝지어 제거하기

def solution(s):
    arr = []
    
    for item in s:
        if arr:
            if arr[-1] == item:
                arr.pop()
            else:
                arr.append(item)
        else:
            arr.append(item)
        
    return 1 if not arr else 0